.github/workflows/agw-workflow.yml

```diff
@@ -0,0 +1,369 @@
+            sudo -E apt-get update -y
+            sudo -E apt-get install -y libsystemd-dev pkg-config curl zip unzip
+            sudo -E apt-get install -y virtualenv python-babel python-dev build-essential python3-setuptools python-setuptools autogen autoconf libtool python3-apt python3-requests python3-pip python-protobuf python3-testresources
+      - name: Setup protoc3
+        run: |
+            pip3 install protobuf
+            pip3 install setuptools==49.6.0
+            curl -Lfs https://github.com/google/protobuf/releases/download/v3.1.0/protoc-3.1.0-linux-x86_64.zip -o protoc3.zip
+            unzip protoc3.zip -d protoc3
+            sudo -E mv protoc3/bin/protoc /bin/protoc
+            sudo -E chmod a+rx /bin/protoc
+            # Workaround: the include files need to be found
+            mv ./protoc3/include/google .
+            sudo -E rm -rf protoc3.zip protoc3
+      - name: Setup Swagger
+        run: |
+            mkdir ${{ env.CODEGEN_ROOT }}
+            wget https://repo1.maven.org/maven2/io/swagger/swagger-codegen-cli/2.2.3/swagger-codegen-cli-2.2.3.jar -O ${{ env.SWAGGER_CODEGEN_JAR }}
+      - name: Execute test_all
+        run: |
+            make -C ${{ env.MAGMA_ROOT }}/lte/gateway/python test_all
+      - name: Upload Test Results
+        if: always()
+        uses: actions/upload-artifact@v2
+        with:
+          name: Unit Test Results
+          path: /var/tmp/test_results
+      - uses: codecov/codecov-action@v1
+        with:
+          files: '/var/tmp/codecovs/cover_lte.xml'
+          flags: lte-test
+      - uses: codecov/codecov-action@v1
+        with:
+          files: '/var/tmp/codecovs/cover_orc8r.xml'
+          flags: lte-test
+      - name: Extract commit title
+        if: failure() && github.ref == 'refs/heads/master'
+        id: commit
+        run: |
+            str="$(jq '.head_commit.message' $GITHUB_EVENT_PATH)"    # get the head_commit message
+            echo ::set-output name=title::${str%%\\n*} | tr -d '"'
+      - name: Notify failure to slack
+        if: failure() && github.ref == 'refs/heads/master'
+        uses: rtCamp/action-slack-notify@v2.0.2
+        env:
+          SLACK_WEBHOOK: ${{ secrets.SLACK_WEBHOOK_CI }}
+          SLACK_TITLE: "Github action lte-test update failed"
+          SLACK_USERNAME: "AGW workflow"
+          SLACK_MESSAGE: "${{ steps.commit.outputs.title}}"
+          SLACK_ICON_EMOJI: ":boom:"
+          SLACK_COLOR: "#FF0000"
+          SLACK_FOOTER: ' '
+
+  session_manager_test:
+    runs-on: ubuntu-latest
+    env:
+      MAGMA_ROOT: "${{ github.workspace }}"
+    steps:
+      - uses: actions/checkout@v2
+      - uses: satackey/action-docker-layer-caching@v0.0.11
+        continue-on-error: true
+      - name: Build the base c/c++ container
+        run: |
+            cd ${{ env.MAGMA_ROOT }}/lte/gateway/docker/mme
+            docker build --tag magma/c_cpp_build --file Dockerfile.ubuntu20.04 ../../../../
```

Comment:
Why not just specify the path to the Docker file instead of having to cd and then reference the buildcontext as magma root.
TLDR: remove the cd and use the absolute path on Dockerfile.ubuntu20.04 and change the build context from ../../../ to .

Reply:
sure!.. let me do that.. 
I copied it from what the original job had